### PR TITLE
 [Logs+] Change default of `ignore_malformed` to `true` in `logs-*-*` data streams

### LINKED ISSUE #95224: Make default logs-*-* mappings more lenient - low hanging fruits
This task is primarily about changing the `ignore_malformed` default for entire `logs-*-*` data streams to `true`.
The exception is the `@timestamp` field, for which we need to keep the default as `false`.

### PR DESCRIPTION
**Pull request by eyalkoren**
### Description

Closes #95224 

As part of our effort to accept all logs by default, one of the first issues we want to address is the rejection of whole log event documents due to field type not matching the corresponding mapping. 
The intention of this issue is to change the default of `ignore_malformed` to `true` specifically for `logs-*-*` data streams, so that the log event will be indexed and the incorrectly typed field will be ignored. The ignored fields, as well as their values, are available through query, as shown in the added test.

## Checklist
- [x] set `ignore_malformed: true` at index level for `logs-*-*` data streams
- [x] override this setting to the `@timestamp` field, for which it should remain `ignore_malformed: false`
- [x] add an integration test
- [x] add to changelog

### CODE REVIEW

File: x-pack/plugin/core/src/main/resources/logs-mappings.json
Comment (felixbarny) [inline x-pack/plugin/core/src/main/resources/logs-mappings.json:13]: This seems redundant as `data-streams-mappings.json` defines the timestamp

Comment (eyalkoren) [reply x-pack/plugin/core/src/main/resources/logs-mappings.json:13]: Yes, I meant to ask about that, I added the related [check](https://github.com/elastic/elasticsearch/pull/95329/files#diff-4c2bc446708490df863212fbf203e77ed14ec697c2b1ef139c1a4d930916f7caR34), which does seem redundant.
I'll remove

Comment (eyalkoren) [reply x-pack/plugin/core/src/main/resources/logs-mappings.json]: The [test failure](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+part-1/32589/console) demonstrated why this was not redundant after all (although misplaced) - [`DataStreamUpgradeRestIT`](https://github.com/elastic/elasticsearch/blob/228fe1f804d93e5a73f88f148c9eb801959e5f10/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/DataStreamUpgradeRestIT.java#L55) demonstrates a composition of only `logs-settings.json` and `logs-mappings.json`, which requires to make this explicit `@timestamp` exception.
I re-added but as a top-level property

File: x-pack/plugin/core/src/main/resources/logs-mappings.json
Comment (ruflin) [inline x-pack/plugin/core/src/main/resources/logs-mappings.json:7]: Is this needed for a data stream? How is this handled today in templates loaded by packages?

Comment (eyalkoren) [reply x-pack/plugin/core/src/main/resources/logs-mappings.json:7]: It did not require any special handling so far. Now, since we set `ignore_malformed: true` at the [backing indices level](https://github.com/elastic/elasticsearch/pull/95329/files#diff-a073a100cddc5733e0a8581e8e0d5fb0b1c675e6f08633eb58b709a62f69a191R13), we need to make this exception explicitly for `@timestamp`, otherwise it fails [validation](https://github.com/elastic/elasticsearch/blob/da214475a5c9604f4f1a32557f319321eae4cdf1/server/src/main/java/org/elasticsearch/index/mapper/DataStreamTimestampFieldMapper.java#L151-L154) (for a good reason).
In this specific case, we could only update the `logs-mapping`, but then I am not sure how well defined it is to have contradicting mappings within the [logs-template composition](https://github.com/elastic/elasticsearch/blob/a2c8e9c847e1f8bf8d47d2b7af2f4de7d877028d/x-pack/plugin/core/src/main/resources/logs-template.json#L6-L8) of all three sources. 
Besides, is there any harm in making it very explicit here? I assume it can only improve, as it will guard from a similar problem if other data stream mappings set `ignore_malformed` at the index level.

File: modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_logs_datastream_defaults.yml
Comment (felixbarny) [inline modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_logs_datastream_defaults.yml:5]: nit: use some more realistic data stream names, for example: `logs-generic-default`

File: modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_logs_datastream_defaults.yml
Comment (felixbarny) [inline modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_logs_datastream_defaults.yml:71]: nit: a terms query seems more appropriate here

File: modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_logs_datastream_defaults.yml
Comment (felixbarny) [inline modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_logs_datastream_defaults.yml:66]: nit: I think it would be easier to follow to move the assertions for a doc right after the ingestion. something like

index: ...
search: ...

index: ...
search: ...

index: ...
search: ...

**Review by felixbarny — APPROVED** [review]

File: modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_logs_datastream_defaults.yml
Comment (eyalkoren) [reply modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_logs_datastream_defaults.yml:71]: I did that as well during my plays with it, I'll change to that.
Only for my education- is this because of how the search is being executed or only because I am looking for an exact match? If it is the latter, wouldn't `match` with `fuzziness: 0` achieve that?

File: modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_logs_datastream_defaults.yml
Comment (felixbarny) [inline modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_logs_datastream_defaults.yml:52]: nit: potentially more descriptive

```suggestion
          message: 'number_as_string'
```

File: modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_logs_datastream_defaults.yml
Comment (felixbarny) [reply modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_logs_datastream_defaults.yml:71]: It'll be semantically the same, I guess but if you want exact matches, you'd typically use a terms query.

File: docs/changelog/95329.yaml
Comment (ruflin) [inline docs/changelog/95329.yaml:2]: @felixbarny I would like us to have somewhere a longer explanation around this change for our users. So someone upgrading and seeing more data gets accepted understands what happened. Should we just make the changelog slightly longer or have it somewhere in the docs?

File: x-pack/plugin/core/src/main/resources/logs-mappings.json
Comment (ruflin) [reply x-pack/plugin/core/src/main/resources/logs-mappings.json:7]: Thanks for the details. I like that we make it explicit for timestamp.

Templates loaded by packages don't use this template, that is why @gsantoro works on the package part.

**Review by ruflin — COMMENTED** [review]
@felixbarny I do not want to expand the scope of this PR here. But can we keep track somewhere of these changes because I think most of these we should eventually also apply to metrics, traces, etc. for all of the data stream naming scheme.

File: docs/changelog/95329.yaml
Comment (felixbarny) [reply docs/changelog/95329.yaml:2]: I guess it doesn't hurt to add a bit more context here. But we'll probably want to have a similar note for integrations.

Maybe something like:
This is our first step in a journey to make the default mappings for logs more lenient.

Trying to index the wrong data type into a field throws an exception by default, and rejects the whole document. The <<ignore-malformed,`ignore_malformed`>> parameter, if set to `true`, allows the exception to be ignored.

Comment (eyalkoren) [reply docs/changelog/95329.yaml:2]: The eventual [release notes](https://www.elastic.co/guide/en/elasticsearch/reference/current/release-notes-8.7.0.html) do not contain much room for elaborated explanations on each changelog entry.
I will add this description to the PR description instead and we'll look for a better way to make these changes visible to users.

**Review by dakrone — APPROVED** [review]
LGTM

**Comment by felixbarny** [comment]
Let's get in https://github.com/elastic/elasticsearch/issues/95469 before merging this to make it easier for users to opt-out, without having to duplicate the whole default logs template.

**Comment by eyalkoren** [comment]
This issue is currently blocked on https://github.com/elastic/elasticsearch/pull/95481

**Pull request merged**
